10_rest/nickolee/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/anz-bank/go-course/10_rest/nickolee/pkg/puppy/store"
+
+	puppy "github.com/anz-bank/go-course/10_rest/nickolee/pkg/puppy"
+	rest "github.com/anz-bank/go-course/10_rest/nickolee/pkg/puppy/rest"
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var out io.Writer = os.Stdout
+
+var (
+	args = os.Args[1:]
+	// this loads your file once parseFlags() is run
+	// these are methods too prepare a variable. Basically telling kingpin how to read the flag from cli
+	fileName = kingpin.Flag("data", "This flag supplies a puppy JSON file name").
+			Short('d').Default("../../puppy-data/puppies.json").File()
+	port   = kingpin.Flag("port", "Port to listen on and serve").Short('p').Default("7777").Int()
+	storer = kingpin.Flag("store", "Map/Sync Store").Short('s').Default("map").Enum("map", "sync")
+)
+
+// Config contains arguments parsed from commandline flags
+type Config struct {
+	file   *os.File
+	port   int
+	storer string
+}
+
+func main() {
+	// parse flags/check if user even provided any
+	cliConfig, err := parseFlags(args)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// read JSON from file + parse/unmarshal the json and we now have []puppy (Go objects)
+	puppies, err := readFileAndUnmarshalPuppies(cliConfig.file)
```

Comment:
Move this block closer to where the puppies are loaded into the store, It's a little weird doing other stuff in the middle that are completely unrelated to these two tasks.